nucypher/network/status_app/crawler.py

```diff
@@ -0,0 +1,184 @@
+    # +-----------+--------+-+---------+-+---------+
+    # |measurement|,tag_set| |field_set| |timestamp|
+    # +-----------+--------+-+---------+-+---------+
+    DB_MEASUREMENT = 'moe_network_info'
+    DB_LINE_PROTOCOL = '{measurement},staker_address={staker_address} ' \
+                           'worker_address="{worker_address}",' \
+                           'start_date={start_date},' \
+                           'end_date={end_date},' \
+                           'stake={stake},' \
+                           'locked_stake={locked_stake},' \
+                           'current_period={current_period}i,' \
+                           'last_confirmed_period={last_confirmed_period}i ' \
+                       '{timestamp}'
+    DB_NAME = 'network'
+
+    DB_RETENTION_POLICY_NAME = 'network_info_retention'
+    DB_RETENTION_POLICY_PERIOD = '5w'  # 5 weeks of data
+    DB_RETENTION_POLICY_REPLICATION = '1'
+
+    def __init__(self,
+                 moe: Moe,
+                 refresh_rate=DEFAULT_REFRESH_RATE,
+                 restart_on_error=True):
+
+        self._moe = moe
+        self._refresh_rate = refresh_rate
+        self._node_learning_task = task.LoopingCall(self._learn_about_nodes)
+        self._contract_learning_task = task.LoopingCall(self._learn_about_contracts)
+        self._restart_on_error = restart_on_error
+        self.log = Logger('moe-crawler')
+
+        # initialize InfluxDB
+        self._client = InfluxDBClient(host='localhost', port=8086, database=self.DB_NAME)
+        self._ensure_db_exists()
+
+        self.__snapshot = dict()
+
+    @property
+    def snapshot(self):
+        return self.__snapshot
+
+    def _ensure_db_exists(self):
+        db_list = self._client.get_list_database()
+        found_db = (list(filter(lambda db: db['name'] == self.DB_NAME, db_list)))
+        if len(found_db) == 0:
+            # db not previously created
+            self.log.info(f'Database {self.DB_NAME} not found, creating it')
+            self._client.create_database(self.DB_NAME)
+            # TODO: review defaults for retention policy
+            self._client.create_retention_policy(name=self.DB_RETENTION_POLICY_NAME,
+                                                 duration=self.DB_RETENTION_POLICY_PERIOD,
+                                                 replication=self.DB_RETENTION_POLICY_REPLICATION,
+                                                 database=self.DB_NAME,
+                                                 default=True)
+        else:
+            self.log.info(f'Database {self.DB_NAME} already exists, no need to create it')
+
+    def _learn_about_contracts(self):
+        period_range = range(1, 365+1)
+        token_counter = {day: NU.from_nunits(self._moe.staking_agent.get_all_locked_tokens(day)).to_tokens()
+                         for day in period_range}
+        self.__snapshot['future_locked_tokens'] = token_counter
```

Comment:
Clever! 🤓 . Alternatively, it could be stored in a different table in the DB if you like - then you could do things like, highest vs lowest periods etc., averages (and periods below and above the average) - may/may not be overkill haha.